frontend/src/components/AuthForms.tsx

```diff
@@ -0,0 +1,286 @@
+import { Button } from "@/components/ui/button";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { loginSchema, signupSchema } from "@/schemas/authSchemas";
+import postsService from "@/services/postsService";
+import { useAuthStore } from "@/stores/authStore";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useMutation } from "@tanstack/react-query";
+import { AxiosError } from "axios";
+import { UUID } from "crypto";
+import { Eye, EyeOff } from "lucide-react";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
+type LoginFormData = {
+  username: string;
+  pass: string;
+};
+
+type SignupFormData = {
+  username: string;
+  email: string;
+  pass: string;
+};
```

Comment:
What is the difference between type and interface???? Ahahaha
just kidding - it's not a necessary thing, but we can change it to be interfaces